### PR TITLE
fix TestReplicaDataIterator Issue #2851

### PR DESCRIPTION
**Pull request by lizhongz**
- Change the order of two raft log keys.
- Check if there is a tombstone key after destroying a range.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/2959)

### CODE REVIEW

File: storage/replica_data_iter_test.go [hunk 1]
```diff
@@ -196,7 +194,16 @@ func TestReplicaDataIterator(t *testing.T) {
...
+		// If the range is destroyed, only a tombstone key will be there.
+		k1, _, _, err := engine.MVCCDecodeKey(iter.Key())
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		iter.Next()
+		if bytes.Compare(k1, keys.RaftTombstoneKey(tc.rng.Desc().RangeID)) != 0 || iter.Valid() {
+			t.Errorf("expected only a tombstone key")
+		}
```
Comment (tschottdorf) [inline storage/replica_data_iter_test.go]: verify here that `!iter.Valid()`.

**Comment by tschottdorf** [comment]
LGTM after my comment.

**Comment by lizhongz** [comment]
Does anyone know what is the error.
```
runTopLevelLifecycles     \ |##############################################---|
npm ERR! Linux 3.14.28-031428-generic
npm ERR! argv "/usr/bin/nodejs" "/go/src/github.com/cockroachdb/cockroach/ui/node_modules/.bin/npm" "install"
npm ERR! node v5.0.0
npm ERR! npm  v3.3.5
npm ERR! path /go/src/github.com/cockroachdb/cockroach/ui/node_modules/strip-indent/node_modules/get-stdin

npm ERR! code ENOENT

npm ERR! errno -2

npm ERR! syscall rename



npm ERR! enoent ENOENT: no such file or directory, rename '/go/src/github.com/cockroachdb/cockroach/ui/node_modules/strip-indent/node_modules/get-stdin' -> '/go/src/github.com/cockroachdb/cockroach/ui/node_modules/strip-indent/node_modules/get-stdin'

npm ERR! enoent This is most likely not a problem with npm itself

npm ERR! enoent and is related to npm not being able to find a file.

npm ERR! enoent 

npm ERR! Please include the following file with any support request:

npm ERR!     /go/src/github.com/cockroachdb/cockroach/ui/npm-debug.log

Makefile:82: recipe for target 'npm.installed' failed

make: *** [npm.installed] Error 254
```

**Comment by tamird** [comment]
That's a circle CI cache issue. I've restarted this build without the cache, that error should go away.

File: storage/replica_data_iter_test.go [hunk 2]
```diff
@@ -196,7 +194,18 @@ func TestReplicaDataIterator(t *testing.T) {
...
-		t.Errorf("expected empty iteration; got first key %q", iter.Key())
+		// If the range is destroyed, only a tombstone key will be there.
+		k1, _, _, err := engine.MVCCDecodeKey(iter.Key())
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		iter.Next()
+		if bytes.Compare(k1, keys.RaftTombstoneKey(tc.rng.Desc().RangeID)) != 0 || iter.Valid() {
+			t.Errorf("expected only a tombstone key")
```
Comment (tamird) [inline storage/replica_data_iter_test.go]: this error message makes it difficult to tell which condition failed.

**Comment by lizhongz** [comment]
@tamird thank you! I've changed the error message which is better now.

File: storage/replica_data_iter_test.go [hunk 3]
```diff
@@ -196,7 +194,18 @@ func TestReplicaDataIterator(t *testing.T) {
...
-		t.Errorf("expected empty iteration; got first key %q", iter.Key())
+		// If the range is destroyed, only a tombstone key will be there.
+		k1, _, _, err := engine.MVCCDecodeKey(iter.Key())
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		iter.Next()
+		if bytes.Compare(k1, keys.RaftTombstoneKey(tc.rng.Desc().RangeID)) != 0 || iter.Valid() {
+			t.Errorf("expected a destroyed replica to have only a tombstone key, but found more")
```
Comment (tamird) [inline storage/replica_data_iter_test.go]: sorry, i wasn't clear before; this error message should be broken into two separate errors:
```go
if tombstoneKey := keys.RaftTombstoneKey(tc.rng.Desc().RangeID); !bytes.Equal(k1, tombstoneKey) {
  t.Errorf("expected tombstone key %q but found %q", tombstoneKey, k1)
}

if iter.Valid() {
  t.Error("expected a destroyed replica to have only a tombstone, but found more")
}
```

**Comment by lizhongz** [comment]
@tamird It's done :-)

**Comment by tschottdorf** [comment]
Thanks @lizhongz! Your contributions are appreciated.

**Comment by lizhongz** [comment]
You are welcome! I am happy to contribute.

**Pull request merged**